src/command_modules/azure-cli-network/azure/cli/command_modules/network/_params.py

```diff
@@ -318,7 +320,8 @@ def completer(prefix, action, parsed_args, **kwargs): # pylint: disable=unused-a
 
 register_cli_argument('network nic create', 'enable_ip_forwarding', options_list=('--ip-forwarding',), help='Enable IP forwarding.', action='store_true')
 register_cli_argument('network nic create', 'network_interface_name', nic_type, options_list=('--name', '-n'), id_part=None, validator=process_nic_create_namespace)
-register_cli_argument('network nic create', 'private_ip_address_version', help='The private IP address version to use.')
+if supported_api_version(ResourceType.MGMT_NETWORK, min_api='2016-09-01'):
+    register_cli_argument('network nic create', 'private_ip_address_version', help='The private IP address version to use.', default=IPVersion.ipv4.value)
```

Comment:
The problem with this approach is that this parameter is still going to appear in older API versions (it just won't have a default value or any help).  These either need to look like this:

```Python
if supported_api_version(ResourceType.MGMT_NETWORK, min_api='2016-09-01'): 
   register_cli_argument('network nic create', 'private_ip_address_version', help='The private IP address version to use.', default=IPVersion.ipv4.value) 
else:
   register_cli_argument('network nic create', 'private_ip_address_version', ignore_type)
```
Or this:
```Python
with VersionConstraint(ResourceType.MGMT_NETWORK, min_api='2016-09-01') as c: 
    c.register_cli_argument('network nic create', 'private_ip_address_version', help='The private IP address version to use.', default=IPVersion.ipv4.value) 
```

My preference would be for the second approach.